src/client/components/HealthCareApp.jsx

```diff
@@ -109,18 +107,30 @@ class HealthCareApp extends React.Component {
       if (!response.ok) {
         throw new Error(response.statusText);
       }
-      store.dispatch(updateSubmissionStatus('submitSucceeded', response.json()));
+      response.json().then(data => {
+        this.props.onUpdateSubmissionStatus('applicationSubmitted', data);
+        console.log('data', data);
+        this.props.onUpdateSubmissionId(data.formSubmissionId);
+        this.props.onUpdateSubmissionTimestamp(data.timeStamp);
+      });
+      setTimeout(() => {
+        this.context.router.push(this.getUrl('next'));
+        this.scrollToTop();
+      }, 8000);
     }).catch(error => {
-      store.dispatch(updateSubmissionStatus('submitFailed', error));
+      setTimeout(() => {
+        this.props.onUpdateSubmissionStatus('submitFailed', error);
+      }, 5000);
```

Comment:
I feel like something else should happen besides just changing the button text if the submission fails. Perhaps a message should appear below the buttons explaining that their submission failed and why it might have failed? At first I thought perhaps the submitMessage component should handle both cases and either way you should be redirected to that page, either with details of your success or something about how your application could not be submitted at this time. But I also see value of staying on the same page because otherwise they will lose all of their data. Not sure what the best solution is here, but some other messaging would probably help.